x86/proofs/simulator.ml

```diff
@@ -1274,19 +1275,55 @@ let run_random_memopsimulation() =
   let l = length icodes in
   let _ = assert (l >= 2) in
   let memop_index = if l >= 6 then l - 4 else l - 2 in
-  cosimulate_instructions (Some memop_index) add_assum icodes;;
+  cosimulate_instructions (Some memop_index)  (if add_assum then 16 else 0) icodes;;
+
+(* ------------------------------------------------------------------------- *)
+(* Cosimulation of simple memory instructions with uniform [rsp+32] address  *)
+(* ------------------------------------------------------------------------- *)
+
+let simple_memory_iclasses = iclasses_simplemem @
+[
+  (*** these were added manually for extra checks ***)
+
+  [0x66; 0x0f; 0x3a; 0x22; 0x7c; 0x24; 0x07; 0x04];
+  [0x66; 0x44; 0x0f; 0x3a; 0x22; 0x74; 0x24; 0x06; 0x63];
+  [0x66; 0x48; 0x0f; 0x3a; 0x22; 0x44; 0x24; 0x05; 0x2a];
+  [0x66; 0x4c; 0x0f; 0x3a; 0x22; 0x44; 0x24; 0x04; 0x0d];
+  [0xc4; 0xe3; 0x69; 0x22; 0x4c; 0x24; 0x07; 0xff];
+  [0xc4; 0x63; 0x79; 0x22; 0x74; 0x24; 0x06; 0x0c];
+  [0xc4; 0xe3; 0x99; 0x22; 0x74; 0x24; 0x05; 0x0b];
+  [0xc4; 0x63; 0xb9; 0x22; 0x44; 0x24; 0x04; 0x05];
+  [0xc4; 0x63; 0x79; 0x16; 0x5c; 0x24; 0x03; 0x13];
+  [0xc4; 0xe3; 0x79; 0x16; 0x4c; 0x24; 0x02; 0x07];
+  [0xc4; 0xe3; 0xf9; 0x16; 0x74; 0x24; 0x01; 0x13];
+  [0xc4; 0x63; 0xf9; 0x16; 0x3c; 0x24; 0x07]
+];;
+
+let simplemem_iclasses =
+  map (fun l -> (fun () -> [l]),true) simple_memory_iclasses;;
+
+let run_random_simplememopsimulation() =
+  let deferred_icodes,add_assum =
+    el (Random.int (length simplemem_iclasses)) simplemem_iclasses in
+  let icodes = deferred_icodes() in
+  let memop_index = 0 in
+  cosimulate_instructions (Some memop_index) 32 icodes;;
 
 (* ------------------------------------------------------------------------- *)
 (* Keep running tests till a failure happens then return it.                 *)
 (* ------------------------------------------------------------------------- *)
 
 let run_random_simulation() =
-  if Random.int 100 < 90 then
+  let rn = Random.int 99 in
```

Comment:
I also have some question about this fraction. I think once we report the number of instructions tested for this new iclass, then we can decide if it is a reasonable fraction or if we should tune it down.